### PR TITLE
Completed CustomConfigurationAttribute

### PR DESCRIPTION
**Pull request by zairoff**

### CODE REVIEW

File: src/Module6/CustomConfiguration/Enum/ProviderType.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace CustomConfiguration.Enum
+{
+    public enum ProviderType
+    {
+        File = 0,
```
Comment (vladislavtrayan) [inline src/Module6/CustomConfiguration/Enum/ProviderType.cs:5]: ProviderType.File never used or implemented - remove it then

File: src/Module6/CustomConfigurationTest/Integration/ConfigurationManagerTests.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using CustomConfiguration;
+using CustomConfigurationTest.Modules;
+using Xunit;
+
+namespace CustomConfigurationTest.Integration
+{
+    public class ConfigurationManagerTests
+    {
+        [Fact]
+        public void LoadConfiguration()
```
Comment (vladislavtrayan) [inline src/Module6/CustomConfigurationTest/Integration/ConfigurationManagerTests.cs:10]: The main idea of unit tests - it should not be dependent on anything, in your case TestConfiguration depends on "C:\Projects\SMT\src\SMT.Api\appsettings.json" , therefore tests are failing on my machine.

File: src/Module6/CustomConfiguration/ConfigurationProviders/JsonConfigurationProvider.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            {
+                result.Add(item.Key, item.Value);
+            }
+
+            return result;
+        }
+
+        public void SaveConfiguration(Dictionary<string, object> keyValuePairs)
+        {
+            throw new NotImplementedException();
```
Comment (vladislavtrayan) [inline src/Module6/CustomConfiguration/ConfigurationProviders/JsonConfigurationProvider.cs:39]: Should be implemented according to task

File: src/Module6/CustomConfiguration/Enum/ProviderType.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+        Xml = 1,
+        Json = 2
+    }
```
Comment (vladislavtrayan) [inline src/Module6/CustomConfiguration/Enum/ProviderType.cs:8]: According to task, you should support AppConfig reading/writing , can't find such functionality in your code

File: src/Module6/CustomConfigurationTest/Integration/ConfigurationManagerTests.cs (hunk 2 above)
Comment (vladislavtrayan) [reply src/Module6/CustomConfigurationTest/Integration/ConfigurationManagerTests.cs:10]: Consider making unit tests independent!

File: src/Module6/CustomConfiguration/ConfigurationProviders/XmlConfigurationProvider.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    public class XmlConfigurationProvider : IConfigurationProvider
+    {
+        private readonly string _path;
+
+        public XmlConfigurationProvider(string path)
+        {
+            _path = path;
+        }
+
+        public Dictionary<string, object> LoadConfiguration()
```
Comment (vladislavtrayan) [inline src/Module6/CustomConfiguration/ConfigurationProviders/XmlConfigurationProvider.cs:16]: If you want support XML format - implement all methods in XmlProvider!

File: src/Module6/CustomConfigurationTest/Unit/ConfigurationManagerTests.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace CustomConfigurationTest.Unit
+{
+    public class ConfigurationManagerTests
```
Comment (vladislavtrayan) [inline src/Module6/CustomConfigurationTest/Unit/ConfigurationManagerTests.cs:14]: Remove class

File: src/Module6/CustomConfiguration/BaseConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,6 @@
+﻿namespace CustomConfiguration
+{
+    public class BaseConfiguration
```
Comment (vladislavtrayan) [inline src/Module6/CustomConfiguration/BaseConfiguration.cs:3]: Should be abstract

**Review by vladislavtrayan — CHANGES_REQUESTED** [review]
Also - make each task in separate branch , it's bad practice to put all tickets changes into one branch!

File: src/Module6/CustomConfigurationTest/Integration/ConfigurationManagerTests.cs (hunk 2 above)
Comment (zairoff) [reply src/Module6/CustomConfigurationTest/Integration/ConfigurationManagerTests.cs:10]: It's integration test

**Comment by vladislavtrayan** [comment]
Approved

**Pull request merged**